lib/srv/server/azure_installer.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/gravitational/trace"
+	"golang.org/x/sync/errgroup"
+
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/cloud/azure"
+)
+
+// AzureInstallerConfig represents configuration for an Azure install script
+// executor.
+type AzureInstallerConfig struct {
+	Emitter     apievents.Emitter
+	AccessPoint auth.DiscoveryAccessPoint
+}
+
+// AzureInstaller handles running commands that install Teleport on Azure
+// virtual machines.
+type AzureInstaller struct {
+	AzureInstallerConfig
```

Comment:
Don't use embedding unless it's an intentional design choice. This makes AzureInstaller now expose all AzureInstallerConfig fields and methods, "dirtying" the API.

I suggest either:

1. Name the field and make it private, keep the constructor below; or
2. Name the field and make it public, ditch the constructor below and let callers use struct initialization. (preferred imo)